src/env/feedback_env.py

```diff
@@ -0,0 +1,149 @@
+from typing import Optional
+
+import gymnasium as gym
+import numpy as np
+
+from src.dataset.custom_feedback_verifier import (
+    RandomFeedback,
+    RuleFeedback,
+    TaskFeedback,
+)
+
+
+class FeedbackEnv:
+    def __init__(self, env: gym.Env, feedback_mode: Optional[str]) -> None:
+        self.env = env
+        self.feedback_mode = feedback_mode
+        if self.feedback_mode:
+            self.env.reset()
+            self.rule_fv = RuleFeedback()
+            self.task_fv = TaskFeedback(self.env)
+            self.random_fv = RandomFeedback(
+                "lorem_ipsum" if "lorem_ipsum" in self.feedback_mode else "random_sentence"
+            )
+
+    def get_base_env(self):
+        return self.env
+
+    def rule_feedback(self, action):
+        return (
+            self.rule_fv.verify_feedback(self.env, action)
+            if self.feedback_mode in ["all", "rule_only"]
```

Comment:
Can we please change this to != "task_only"? As we need this type of feedback for numerical and random feedback too.